### PR TITLE
Make calculate new vl configurable

### PR DESCRIPTION
**Pull request by rez5427**
I don't know whether sail-riscv needs this.
The spike uses VLMAX to calculate new vl, and sail-riscv uses (AVL + 1) / 2.

The [spec](https://drive.google.com/file/d/1uviu1nH-tScFfgrovvFCrj7Omv8tFtkp/view?usp=drive_link) says:
> The number of bits implemented in vl depends on the implementation’s maximum vector
> length of the smallest supported type. The smallest vector implementation with VLEN=32
> and supporting SEW=8 would need at least six bits in vl to hold the values 0-32
> (VLEN=32, with LMUL=8 and SEW=8, yields VLMAX=32).

### CODE REVIEW

**Comment by Timmmm** [comment]
LGTM but I would add some explanation and move the if to the place where it is actually used. Also move the `to_bits` outside the body to make it clearer.

```
val sys_vext_vl_use_ceil = pure "sys_vext_vl_use_ceil" : unit -> bool

function calculate_new_vl(AVL : int, VLMAX : int) -> xlenbits = {
  // See "Constraints on Setting vl" in the vector spec.
  let vl =
    if AVL <= VLMAX then AVL
    else if AVL < 2 * VLMAX then {
      // If VLMAX < AVL < 2 * VLMAX then we can use any value
      // such that ceil(AVL / 2) <= vl <= VLMAX. Here we provide
      // two options: ceil(AVL / 2) or VLMAX.
      if sys_vext_vl_use_ceil() then (AVL + 1) / 2 else VLMAX
    }
    else VLMAX;

  to_bits(xlen, vl)
}
```

(Feel free to copy/paste.)

**Review by Timmmm — APPROVED** [review]
LGTM! I'm not a vector expert though so we might want to get the opinion of someone who is (although the spec seems pretty unambiguous on this point). I can ask a colleague if nobody is forthcoming.

**Comment by avakar** [comment]
@Timmmm's colleague here. LGTM.

File: c_emulator/riscv_platform_impl.c [hunk 1]
```diff
@@ -32,6 +32,8 @@ uint64_t rv_ram_size = UINT64_C(0x4000000);
 uint64_t rv_rom_base = UINT64_C(0x1000);
 uint64_t rv_rom_size = UINT64_C(0x100);
 
+bool rv_vext_vl_use_ceil = false;
```
Comment (jrtc27) [inline c_emulator/riscv_platform_impl.c:35]: The commit message doesn't mention that you're changing the default behaviour

Comment (rez5427) [reply c_emulator/riscv_platform_impl.c:35]: Sorry..., Is this commit message ok?

Comment (Timmmm) [reply c_emulator/riscv_platform_impl.c:35]: I'm not sure which commit message you changed... but don't worry I will squash it and write a descriptive message.

**Pull request merged**